create light-client-updates bucket

Inspector-Butters opened the pull request:
**What type of PR is this?**
 Feature

**What does this PR do? Why is it needed?**
This PR intends to address the needs for a light client updates database per #12991 .
This is needed to support light clients. 

**Which issues(s) does this PR fix?**
Partially addresses #12991

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/prysmaticlabs/prysm?pullRequest=14266) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/prysmaticlabs/prysm?pullRequest=14266) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/prysmaticlabs/prysm?pullRequest=14266) it.</sub>

Inspector-Butters commented:
@rkapka 
- added a `LightClientUpdateWithVersion` struct to the protos to save in the db.
- added `SaveLightClientUpdate`, `LightClientUpdate`, and `LightClientUpdates`.
- stuck at writing the tests, because I can't import functions defined in `blockchain/lightclient.go`. (import cycle error). any help is appreciated.
- added  beaconDB to the lightclient server in `rpc/eth/lightclient/server.go` so we can read the updates in the handlers later on.

rkapka commented on beacon-chain/db/kv/lightclient.go:
```diff
@@ -0,0 +1,65 @@
+	return &update, err
+}
+
+func uint64ToBytes(period uint64) []byte {
```
You should use the existing `bytesutil.Uint64ToBytesBigEndian` function

rkapka commented on beacon-chain/db/kv/lightclient.go:
```diff
@@ -0,0 +1,65 @@
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.LightClientUpdates")
+	defer span.End()
+
+	var updates []*ethpbv2.LightClientUpdateWithVersion
```
This is OK, but as a small optimization you can pre-allocate the capacity of the slice, since you know the slice won't be bigger than the number of periods.
```
updates := make([]*ethpbv2.LightClientUpdateWithVersion, 0, endPeriod - startPeriod + 1)
```

rkapka commented on beacon-chain/db/kv/lightclient.go:
```diff
@@ -0,0 +1,65 @@
+	})
+}
+
+func (s *Store) LightClientUpdates(ctx context.Context, startPeriod, endPeriod uint64) ([]*ethpbv2.LightClientUpdateWithVersion, error) {
```
It might be worth adding a defensive check that `endPeriod >= startPeriod`

Inspector-Butters commented:
@rkapka one thing, do we want to have safety guards in the kv interface functions that check that an update is somewhat valid before saving it? for example, the important fields are not nil.

rkapka commented on beacon-chain/db/kv/lightclient_test.go:
```diff
@@ -0,0 +1,36 @@
+		SyncAggregate:           nil,
+		SignatureSlot:           7,
+	}
+	require.NotNil(t, update, "update is nil")
```
This is unnecessary since we created the object by hand.

rkapka reviewed (commented):
We should have a few test cases for `LighClientUpdates`:
- `startPeriod > endPeriod` - error returned
- `startPeriod == endPeriod` - one update returned
- `startPeriod > endPeriod`:
  - all updates between `startPeriod` and `endPeriod` are in the DB - all returned
  - some periods between `startPeriod` and `endPeriod` don't have updates - existing updates returned
  - `startPeriod` is earlier than the first update - return starting from first existing update
  - `endPeriod` is later than the first update - return up to the last update

Inspector-Butters commented:
The `kv.LightClientUpdates()` now returns all the available updates in the requested range in a map.

rkapka commented on beacon-chain/db/kv/lightclient.go:
```diff
@@ -0,0 +1,79 @@
+
+		firstPeriodInDb, _ := c.First()
+		if firstPeriodInDb == nil {
+			return fmt.Errorf("no light client updates in the database")
```
We can return an empty map, not having any updates is not an error
